lib/chef/resource/plist.rb

```diff
@@ -189,7 +189,12 @@ def plistbuddy_command(subcommand, entry, path, value = nil)
         sep = " "
         arg = case subcommand.to_s
               when "add"
-                type_to_commandline_string(value)
+                if value.is_a?(Hash)
+                  sep = ":"
+                  value.map { |k, v| "#{k} #{type_to_commandline_string(v)}" }.join(sep)
```

Comment:
I don't think you need the `.join(sep)` here?  It isn't in the other branch, and it looks like it falls through and will hit the `.join(sep).strip` after the case statement.

Reply:
Yep, you're right. Missed that when porting this from one of our other libraries.